enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -183,41 +196,25 @@ Currently developing on a feature branch on the cluster-capi-operator [here](htt
 
 ### Risks and Mitigations
 
-N/A
+A risk of using transport ConfigMaps to produce and load CAPI providers components is the hard limit on the ConfigMap size in Kubernetes.
+In fact Kubernetes has a maximum size of 1MiB for a ConfigMap. If the manifest exceeds this size, Kubernetes will throw an error on manifest creation.
+To mitigate this we can address this from two angles:
+- early prevention of this issue by checking the size of the ConfigMap before writing it to disk by the manifest generator, and erroring if that exceeds the maximum allowed size
+- in the future, if this issue arises, leverage the same mechanism the upstream cluster-api-operator uses and compress/decompress the ConfigMap (see the [upstream specification](https://github.com/kubernetes-sigs/cluster-api-operator/blob/93989812d7d15f3b5dd1930bef9e0208b7301cee/docs/README.md?plain=1#L693-L729) for more details)
```

Comment:
What about sharding? We could also shard the configmap into multiple configmaps right?

Reply:
Any risk of exceeding the limit even when compressed? What drives growth in size here?

Reply:
@JoelSpeed  Yes sharding the ConfigMap with the right labelling is also an option, now that we apply/update all the components on a single reconcile.
I was suggesting compressing as that is what upstream does. I think we could keep both as potential options to take on if we ever encountered size issues.

@sdodson to your point, the biggest ConfigMap I found for a provider is the AWS one with 626 KB, against the 1 MiB limit. Growth is mainly driven by new CRDs introduced by the provider. 
Taking a look at the effectiveness of the compression described in the upstream approach [1] it looks like we the compressed manifests are **almost one order of magnitude lower in size than the original one**, which I think gives us plenty of space for growth.
 
[1]: compression test
```
[~/src/cluster-api-provider-aws/openshift] $ gzip -c infrastructure-components.yaml > components.gz
[~/src/cluster-api-provider-aws/openshift] $ ls -lh *components*
-rw-r--r--  1 ddonati  staff    76K 27 Sep 10:45 components.gz
-rw-r--r--  1 ddonati  staff   626K 19 Sep 14:16 infrastructure-components.yaml
```